source/runtime.cpp

```diff
@@ -1143,7 +1132,12 @@ void reshade::runtime::save_screenshot(const std::wstring &postfix, const bool s
 	}
 	else if (_screenshot_include_preset && should_save_preset)
 	{
-		save_preset(least + L".ini");
+		std::error_code ec;
+
+		if (ini_file::load_cache(_current_preset_path).flush())
```

Comment:
I think I'd prefer there to be an `ini_file::flush_cache(path)` function which does this instead. That way `ini_file::flush` can be made private (like `ini_file::save` was before).